### PR TITLE
[BB2] FAQ

### PR DESCRIPTION
**Pull request by klshuster**
**Patch description**
I've included a README for BB2 with FAQ that I've seen in our issues. Hopefully we can direct people towards this if it has the answer...

### CODE REVIEW

File: projects/blenderbot2/agents/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+IndexError: index 1 is out of bounds for dimension 0 with size 1
+```
+If you notice this during training, please set `--memory-decoder-model-file ''`.
+
+### How can I use BlenderBot2 with **only** search?
+
+You'll need to do two things:
+
+1. Set `--knowledge-access-method search_only`
+2. Set `--query-generator-model-file zoo:sea/bart_sq_gen/model`
```
Comment (mojtaba-komeili) [inline projects/blenderbot2/agents/README.md:46]: Is this one necessary? Isn't that the default atm?

File: projects/blenderbot2/agents/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+### How can I train with gold documents provided?
```
Comment (mojtaba-komeili) [inline projects/blenderbot2/agents/README.md:48]: Should we add a note here that what we mean by gold docs. Maybe to keep it general enough we could say something like this:

Gold documents are any set of documents that you need you retriever to surface. We add them to make sure the generator model sees (is conditioned on) a certain set of document. Example use cases are as follows:
1- Adding documents to the mix of retrieved that you are certain have useful knowledge that the model should use for generating response.
2- Ensure reproducibility between experiments for retrievers that have randomized responses. For example,  if you wanted your model to see the same documents that was shown to crowdsourcing agent 
 (eg `WizIntGoldDocRetrieverFiDAgent` model with `wizard of internet` dataset).

**Review by mojtaba-komeili — APPROVED** [review]
Thanks for adding this. Reading it was useful for me too.

File: projects/blenderbot2/agents/README.md (hunk 1 above)
Comment (klshuster) [reply projects/blenderbot2/agents/README.md:46]: No, this is not the default. The default uses `--knowledge-access-method classify`, with the query generator from BB2 (trained to either search or retrieve from memory)

File: projects/blenderbot2/agents/README.md (hunk 2 above)
Comment (klshuster) [reply projects/blenderbot2/agents/README.md:48]: Great idea, i'll add that note

File: projects/blenderbot2/agents/README.md (hunk 1 above)
Comment (mojtaba-komeili) [reply projects/blenderbot2/agents/README.md:46]: Sorry, I was referring to `--query-generator-model-file zoo:sea/bart_sq_gen/model`. That can be default, right?

Comment (klshuster) [reply projects/blenderbot2/agents/README.md:46]: At the moment it's not, the default is `zoo:blenderbot2/query_generator/model`, which determines whether to search or retrieve from memory

Comment (mojtaba-komeili) [reply projects/blenderbot2/agents/README.md:46]: Oh, yeah, I remember now.

**Pull request merged**